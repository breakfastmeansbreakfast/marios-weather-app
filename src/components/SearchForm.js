import React from "react";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => {
    setSearchText(event.target.value);
  };
  return (
    <div className="search-form">
      <input
        type="text"
        placeholder="Search for a UK town or city"
        onChange={handleInputChange}
        value={searchText}
      />
      <button type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
}

export default SearchForm;
