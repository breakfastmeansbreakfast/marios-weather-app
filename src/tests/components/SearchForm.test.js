import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchText: "Cornwall",
    setSearchText: jest.fn(),
    onSubmit: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders a search box", () => {
    const { getByRole } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(getByRole("textbox")).toBeInTheDocument();
  });

  it("handles event change when button is clicked", () => {
    render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    fireEvent.click(screen.getByRole("button"));
    expect(validProps.onSubmit).toHaveBeenCalled();
  });
});
