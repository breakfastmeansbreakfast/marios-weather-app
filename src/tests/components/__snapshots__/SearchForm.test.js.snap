// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      placeholder="Search for a UK town or city"
      type="text"
      value="Cornwall"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
